packages/babel-types/src/definitions/experimental.js

```diff
@@ -269,3 +270,40 @@ defineType("BigIntLiteral", {
   },
   aliases: ["Expression", "Pureish", "Literal", "Immutable"],
 });
+
+defineType("RecordExpression", {
+  visitor: ["properties"],
+  aliases: ["Expression"],
+  fields: {
+    syntaxType: {
+      validate: assertValueType("string"),
```

Comment:
Let's be more explicit
```js
assertOneOf("bar", "hash")
```

Reply:
I'm not sure that this information should be there. This seems like syntax-specific information, not related to the AST.
It's similar to how the presence or absence of a semicolon is not represented in the AST.

I would prefer it as an option in `@babel/generator`, which should throw when printing a `RecordExpression` node if that option is not specified.
`@babel/plugin-syntax-record-and-tuple` would set both the parser option and the generator option.

Reply:
@nicolo-ribaudo do you mean an additional property on the node specifically for the generator to use to pick syntax to output? If so, why wouldn't you just use the existing `syntaxType` property?

Reply:
The generator options is independent on AST. By doing so we can have the AST agnostic about the syntax types, which is semantically equivalent.

You can add a new format option here https://github.com/babel/babel/blob/c35ba3d3abd0b915849be13fc682934c54f41599/packages/babel-generator/src/index.js#L39

and reference this option in the generator via `this.format.recordTupleSyntaxType`.

In the `syntax-*` you can mutate the generator options via `options.generatorOpts.recordTupleSyntaxType`.